### PR TITLE
Oy2 15215 - Deviations from Measure Specifications/OMS logic QA fixes

### PR DESCRIPTION
**Pull request by jdinh8124**
## Purpose
Fix issues found in Padma's QA testing for Deviations from Measure Specifications/OMS logic

#### Linked Issues to Close
https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/208?modal=detail&selectedIssue=OY2-15215

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/measures/2021/FUAAD/index.tsx [hunk 1]
```diff
@@ -84,6 +84,12 @@ export const FUAAD = ({ name, year, handleSubmit }: Measure.Props) => {
           <Q.OptionalMeasureStratification
             ageGroups={ageGroups}
             totalLabel={Q.DefaultOptionalMeasureStratProps.totalLabel}
+            deviationConditions={{
+              show30DaysAges18To64,
+              show30DaysAges65AndOlder,
+              show7DaysAges18To64,
+              show7DaysAges65AndOlder,
+            }}
```
Comment (daniel-belcher) [inline services/ui-src/src/measures/2021/FUAAD/index.tsx:92]: This is a lot simpler than the watchField setup I had been trying, really like it.

File: services/ui-src/src/measures/2021/FUAAD/questions/OptionalMeasureStratification.tsx [hunk 2]
```diff
@@ -691,6 +759,7 @@ export const OptionalMeasureStratification = ({
               <AgeData
                 ageGroups={ageGroups}
                 totalLabel={totalLabel}
+                deviationConditions={deviationConditions}
                 {...register("ACAGroupRate")}
```
Comment (daniel-belcher) [inline services/ui-src/src/measures/2021/FUAAD/questions/OptionalMeasureStratification.tsx:763]: Not a request for change, but a critique of my initial setup, I feel like there should be a better way to pass all of these to the subcomponent.

File: services/ui-src/src/measures/schema.ts [hunk 3]
```diff
@@ -3,15 +3,22 @@ import { Measure } from "./types";
 
 const RateJoiValidator = Joi.array().items(
   Joi.object({
-    numerator: Joi.string(),
-    denominator: Joi.string(),
-    rate: Joi.string(),
+    numerator: Joi.string().empty(),
+    denominator: Joi.string().empty(),
+    rate: Joi.string().empty(),
```
Comment (daniel-belcher) [inline services/ui-src/src/measures/schema.ts:8]: Do we not need to pass anything into the empty() for number fields?

**Review by daniel-belcher — DISMISSED** [review]
This looks good to me.

File: services/ui-src/src/measures/2021/FUAAD/questions/OptionalMeasureStratification.tsx (hunk 2 above)
Comment (jdinh8124) [reply services/ui-src/src/measures/2021/FUAAD/questions/OptionalMeasureStratification.tsx:763]: You're right I'll take a look at it again

Comment (jdinh8124) [reply services/ui-src/src/measures/2021/FUAAD/questions/OptionalMeasureStratification.tsx:763]: Open for ideas though if you think we can streamline this somehow

File: services/ui-src/src/measures/schema.ts (hunk 3 above)
Comment (jdinh8124) [reply services/ui-src/src/measures/schema.ts:8]: Nice catch

**Review by abelzer212 — APPROVED** [review]

**Review by daniel-belcher — APPROVED** [review]
The provider changes helped clean and drop this file down a bit, but it made that git diff way harder to check for other changes. Reading through everything looks good though.

**Comment by jdinh8124** [comment]
> The provider changes helped clean and drop this file down a bit, but it made that git diff way harder to check for other changes. Reading through everything looks good though.

Thanks for the idea! Definitely made things cleaner

**Pull request merged**